workflows/diagnostics/fv3net/diagnostics/offline/compute.py

```diff
@@ -308,9 +310,14 @@ def main(args):
     # add Q2 and total water path for PW-Q2 scatterplots and net precip domain averages
     if any(["Q2" in v for v in model.output_variables]):
         model = fv3fit.DerivedModel(model, derived_output_variables=["Q2"])
-
+    if "ncol" in evaluation_grid.dims:
+        grid_name = "ncol"
+    else:
+        grid_name = "x"
     ds_predicted = get_prediction(
-        config=config, model=model, evaluation_resolution=evaluation_grid.sizes["x"]
+        config=config,
+        model=model,
+        evaluation_resolution=evaluation_grid.sizes[grid_name],
```

Comment:
```suggestion
        evaluation_resolution=evaluation_grid.sizes[horizontal_dims[0]],
```
And infer `horizontal_dims` somewhere earlier in code. And then can also pass `horizontal_dims` to `_compute_diagnostics` instead of inferring it within that function.